libraries/AP_GPS/AP_GPS_GSOF.cpp

```diff
@@ -86,7 +86,7 @@ AP_GPS_GSOF::read(void)
 }
 
 bool
-AP_GPS_GSOF::parse(uint8_t temp)
+AP_GPS_GSOF::parse(const uint8_t temp)
```

Comment:
I'm personally not a fan of adding const to the argument list.  I think it makes it slightly harder to read by making the arg list too long especially for functions with a lot of arguments.  There are varying views in the dev team though.

Reply:
I'll stand up as a "in favor of const in signatures". It's successfully caught me accidentally changing things I hadn't intended to before.

Reply:
I am too - but only in the implementation, *not* in the header file.

I think that gives you short parameter lists in the header file (which is what Randy likes), but enforces the const-ness in the implementation (at the expense of a verbose parameter declaration in the cpp).

Reply:
I'd prefer reducing bug probability over readability. `const` also indicates to the developer, reading the header, that `temp` can't change, so it's a pass-by-const-value rather than pass-by-value, which is semantically different. 

Functions, generally, shouldn't have 15 arguments. After about ~4, one should consider encapsulating them in `struct` or reducing the scope of the function. Since the arg list is the same length, it's really just a line length thing, which we can fix by breaking the arguments to 1 per line.

This also only has 1 argument

Reply:
From the CPP core guidelines: `Interfaces that promise “no change” of objects passed as arguments greatly increase readability.`

Putting it in the implementation removes some of that benefit. Unless you go dig into the implementation, and see that the implementation is different than the prototype.

Reply:
I definitely wouldn't vote for having the "const" there in the implementation but not the header file.  That's doesn't seem like a good idea.

Reply:
I'm perplexed by the arguments against this. These changes help, they *DO NOT* hurt whatsoever. I think yiu guys need to suck it up about not liking to see "const" in headers and all that crap. As @WickedShell said, he's saved him in the past and I know it has helped me. I feel strongly about this so I'm going to override @rmackay9 and @peterbarker 's concerns on this.

The WARN_IF_UNUSED in the header could be handled by the compiler but we use it in lots of other places already so it should go in too.